scripts/install.sh

```diff
@@ -0,0 +1,253 @@
+	edit_json 'this.config["validate-commit-msg"]={
+	"types": [
+    "feat",
+    "fix",
+    "docs",
+    "style",
+    "refactor",
+    "perf",
+    "test",
+    "chore",
+    "revert",
+    "ci",
+    "build"
+  ],
+  "warnOnFail": false,
+  "maxSubjectLength": 100,
+  "subjectPattern": "^[A-Za-z]+-[0-9]+((,| - ).*)?",
+  "subjectPatternErrorMsg": "Wrong commit format! Subject does not match subject pattern!",
+  "helpMessage": ""}'
+}
+
+function edit_json() {
+	# in-place editing, see https://github.com/trentm/json#readme
+	json -I -f package.json -e "$1"
+}
+
+function npm_package_is_installed() {
+  # set to 1 initially
+  local ret=1
+  # set to 0 if not found
+  local pkg=$(echo $1 | cut -d/ -f 1)
+
+  ls node_modules | grep "$pkg" >/dev/null 2>&1||
+  npm list -g --depth=0 | grep "$pkg" >/dev/null 2>&1 || 
+  { local ret=0; }
+  # return value
+  echo "$ret"
+}
+
+function install_node_modules() {
+	npm config set registry="https://registry.npmjs.org"
+	if [ ! -d "${ROOT}/node_modules" ]; then
+		mkdir "${ROOT}/node_modules"
+	fi
+	info "Installing global node dependancies..." 
+	for plugin in ${node_modules[@]}; do 
+		if [ $(npm_package_is_installed $plugin) == 1 ]; then
+		echo "$plugin installed, skipping ..."
+		else
+		  if [ ${OS} = "Ubuntu" ]; then
+		  sudo $INSTALLER install $Global_opt $plugin
+		  else
+	          $INSTALLER install $Global_opt $plugin
+		  fi	
+		fi 
+	done 
+
+	info "Installing local node dev dependancies"
+	for plugin in ${node_dev_modules[@]}; do 
+		if [ $(npm_package_is_installed $plugin) == 1 ]; then
+		echo "$plugin installed, skipping ..."
+		else
+		$INSTALLER install $Dev_opt $plugin 
+		fi 
+	done 
+	info "done."
+}
+
+function install_node_modules_on_MacOSX() {
+	set -e
+	# dealing with mac libtool conflictions
+	# see discussion https://github.com/barrysteyn/node-scrypt/issues/113
+	mv /usr/local/bin/libtool /usr/local/bin/libtool.bak
+	$INSTALLER rebuild node-sass
+	install_node_modules
+	mv /usr/local/bin/libtool.bak /usr/local/bin/libtool
+
+}
+
+function install_pcl() {
+# http://www.pointclouds.org/documentation/tutorials/installing_homebrew.php
+	brew install pcl
+# https://github.com/totakke/homebrew-openni, homebrew/science has been discarded.
+	brew tap brewsci/science
+	brew tap toakke/openni
+	brew install openni
+	brew install sensor
+	brew install sensor-kinect
+	brew install nite
+
+	brew tap toakke/openni2
+	brew install openni2
+	brew install openni2-freenectdriver
+
+}
+
+main() {
+	if [ $(uname -s) == "Darwin" ]; then
+		OS="OSX"
+		info "<$(uname -s)> detected. checking Max OSX versions: Major.Minor ..."
+		info "installing ..."
+		OSX
+
+		configGitCommitter
+
+	else
+		if [ -f /etc/lsb-release ]; then
+		OS="Ubuntu"
+		Ubuntu
+
+		configGitCommitter
+	
+		else
+		warn "platform <$(uname -s)> Not supported yet. Please install the package dependencies manually. Pull requests are welcome!"
+		fi
+	fi 
+
+}
+
+main
```

Comment:
What is the purpose of this file?

Reply:
It prepares many depandencies. In this commit, it just provides me with git commit hook.

Reply:
OK, lets remove it please.

Reply:
copy it.